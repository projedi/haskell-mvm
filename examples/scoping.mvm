int i;
i = 0;
{
  print(i);
  int i;
  i = 1;
  print(i);
}
print(i);

i = 0;
int f() {
  return i + 1;
}
print(f());
i = 1;
print(f());

int j;
int f(int i) {
  return i + j;
}
i = 0;
j = 1;
print(f(3));
i = 10;
j = 20;
print(f(3));

void f() {
  i = i + 1;
}
i = 0;
f();
print(i);
f();
print(i);

{
  int i;
  int j;
  void f() {
    i = i + 3;
  }
  i = 0;
  {
    int i;
    int j;
    i = 10;
    void g() {
      i = i + 1;
      f();
      f();
      print(i);
    }
    g();
  }
  print(i);
}
