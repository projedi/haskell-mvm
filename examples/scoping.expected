100
1
100
1
2
4
23
1
2
11
6
